ui/lib/login.js

```diff
@@ -11,17 +11,38 @@ export function saveToVisitedOrgs(domain, orgName) {
       name: orgName,
     })
 
-    // set the cookie domain to a general base domain
-    let cookieDomain = window.location.host
-    let parts = cookieDomain.split('.')
-    if (parts.length > 2) {
-      parts.shift() // remove the org
-      cookieDomain = parts.join('.') // join the last two parts of the domain
-    }
-
     cookies.set('orgs', visitedOrgs, {
       path: '/',
-      domain: `.${cookieDomain}`,
+      domain: `.${currentBaseDomain()}`,
     })
   }
 }
+
+export function currentBaseDomain() {
```

Comment:
Should use `/api/server-configuration` which has the base domain and fall back to `window.location` if it's empty.

Reply:
Made this change